lms/djangoapps/courseware/model_data.py

```diff
@@ -840,6 +840,8 @@ def set_many(self, kv_dict):
             kv_dict (dict): dict mapping from `DjangoKeyValueStore.Key`s to field values
         Raises: DatabaseError if any fields fail to save
         """
+        if self.read_only:
```

Comment:
For manual testing, I faked the user agent in Chrome to verify that no rows got written in CSM. Also verified that it state continues to be written for non-crawler user agents.